packages/esm-patient-chart-app/src/visit/visits-widget/visit-detail-overview.component.tsx

```diff
@@ -0,0 +1,216 @@
+                        <TableCell key={cell.id}>{cell.value}</TableCell>
+                      ))}
+                    </TableExpandRow>
+                    {row.isExpanded && (
+                      <TableExpandedRow className={styles.expandedRow} colSpan={headers.length + 2}>
+                        <div className={styles.EncounterObservations}>
+                          <EncounterObservations encounterUuid={row.id} />
+                        </div>
+                      </TableExpandedRow>
+                    )}
+                  </React.Fragment>
+                ))}
+              </TableBody>
+            </Table>
+            {encounters.length === 0 && <p className={styles.dataTableRow}>No encounters found.</p>}
+          </TableContainer>
+        );
+      }}
+    </DataTable>
+  );
+};
+
+interface SingleVisitDetailComponentProps {
+  visit: Visit;
+}
+
+const SingleVisitDetailComponent: React.FC<SingleVisitDetailComponentProps> = ({ visit }) => {
+  const { t } = useTranslation();
+  const [listView, setView] = useState<boolean>(true);
+  const encounters = useMemo(
+    () =>
+      visit.encounters.map((encounter: OpenmrsResource, ind) => ({
+        id: encounter.uuid,
+        time: dayjs(encounter.encounterDateTime).format('hh:mm'),
+        encounterType: encounter.encounterType.display,
+        provider: encounter.encounterProviders.length > 0 ? 'Provider' : '',
+      })),
+    [visit],
+  );
+
+  return (
+    <div className={styles.visitsDetailWidgetContainer}>
+      <div className={styles.visitsDetailHeaderContainer}>
+        <h4 className={styles.productiveHeading02}>
+          {visit.visitType.display}
+          <br />
+          <p className={styles.bodyLong01}>{formatDateTime(visit.startDatetime)}</p>
+        </h4>
+        <div className={styles.toggleButtons}>
+          <Button
+            className={`${styles.toggle} ${listView ? styles.toggleActive : ''}`}
+            size="small"
+            kind="ghost"
+            onClick={() => setView(true)}>
+            {t('All Encounters', 'All Encounters')}
+          </Button>
+          <Button
+            className={`${styles.toggle} ${!listView ? styles.toggleActive : ''}`}
+            size="small"
+            kind="ghost"
+            onClick={() => setView(false)}>
+            {t('Visit Summary', 'Visit Summary')}
+          </Button>
+        </div>
+      </div>
+      {listView && visit?.encounters && <EncounterListDataTable encounters={encounters} />}
+    </div>
+  );
+};
+
+interface VisitOverviewComponentProps {
+  patientUuid: string;
+}
+
+function VisitDetailOverviewComponent({ patientUuid }: VisitOverviewComponentProps) {
+  let A = [1, 2, 3, 4, 5];
+
+  const [visits, setvisits] = useState<Array<Visit>>([]);
+
+  useEffect(() => {
+    if (patientUuid) {
+      const abortController = new AbortController();
+      getVisitsForPatient(patientUuid, abortController, 'full').subscribe(({ data }) => {
```

Comment:
Shouldn't that be cleaned up? When is the abort controller used?

Reply:
The function `getVisitsForPatient` is using the abortController to signal if any error pops in, I copied it from the visit dashboard, but I didn't use the abortController anywhere.

Can you guide me how should I use the abortController?

Reply:
I looked into `AbortController` and I don't need the abort controller in my use. 
Should I make my custom `FetchVisits` function instead of using this one, or  keep it as it is?

Reply:
You always should abort / clean up your effect. As it is now this will result in a leak. You don't even unsubscribe here.

Reply:
@FlorianRappl duly noted. I have done the changes in the latest push.